client/src/components/scans/scanFailedHostList.js

```diff
@@ -0,0 +1,205 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { EmptyState, Grid, Icon } from 'patternfly-react';
+import helpers from '../../common/helpers';
+import { connect } from 'react-redux';
+import { getConnectionScanResults, getInspectionScanResults } from '../../redux/actions/scansActions';
+
+class ScanFailedHostList extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      scanResults: [],
+      scanResultsError: false,
+      connectionScanResultsPending: false,
+      inspectionScanResultsPending: false
+    };
+  }
+
+  componentDidMount() {
+    this.refresh();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Check for changes resulting in a fetch
+    if (!_.isEqual(nextProps.lastRefresh, this.props.lastRefresh)) {
+      this.refresh();
+    }
+  }
+
+  isIpAddress(name) {
+    let vals = name.split('.');
+    if (vals.length === 4) {
+      return _.find(vals, val => Number.isNaN(val)) === undefined;
+    }
+    return false;
+  }
+
+  ipAddressValue(name) {
+    const values = name.split('.');
+    return values[0] * 0x1000000 + values[1] * 0x10000 + values[2] * 0x100 + values[3] * 1;
+  }
+
+  addResults(results) {
+    const { scanResults } = this.state;
+
+    let failedResults = [];
+    _.forEach(_.get(results, 'value.data.results', []), result => {
+      if (result.status === 'failed') {
+        failedResults.push(result);
+      }
+    });
+
+    let allResults = [...scanResults, ...failedResults];
+
+    allResults.sort((item1, item2) => {
+      if (this.isIpAddress(item1.name) && this.isIpAddress(item2.name)) {
+        const value1 = this.ipAddressValue(item1.name);
+        const value2 = this.ipAddressValue(item2.name);
+
+        return value1 - value2;
+      }
+
+      return item1.name.localeCompare(item2.name);
+    });
```

Comment:
Optional: Moving the IP related functionality to the helper/common area
Note: Getting weird results on the sort with the `localeCompare`, still looks alphabetized on mine... trying to figure it out, can't tell if it's my settings or the method. Running the sort off to the side independently works fine

![screen shot 2018-03-21 at 3 00 00 pm](https://user-images.githubusercontent.com/3761375/37731843-b4391ae0-2d19-11e8-910d-23f643764e57.png)

Reply:
Are you looking in the Scans view at failed hosts? The sources view failed hosts do not sort because we are using server side paging/sorting. No sense sorting the local results when it could be just a subset.